web/src/lib/components/pages/record/profile-actions.tsx

```diff
@@ -0,0 +1,84 @@
+import { h, Component } from 'preact';
+import Profile from '../profile/profile';
+import User from '../../../user';
+
+interface Props {
+  user: User;
+}
+
+interface State {
+  profileFormVisible: boolean;
+  whyProfileVisible: boolean;
+}
+
+export default class ProfileActions extends Component<Props, State> {
+  state: State = {
+    profileFormVisible: false,
+    whyProfileVisible: false,
+  };
+
+  private toggleProfileForm = () => {
+    this.setState({ profileFormVisible: !this.state.profileFormVisible });
+  };
+
+  private toggleWhyProfile = () => {
+    this.setState({ whyProfileVisible: !this.state.whyProfileVisible });
+  };
+
+  render() {
+    return (
+      <div id="profile-actions">
+        <hr />
+        {this.state.profileFormVisible ? (
+          <div id="profile-form-container">
+            <a class="cancel" onClick={this.toggleProfileForm}>
+              Cancel
+            </a>
+            <Profile active="" user={this.props.user} />
+          </div>
+        ) : this.props.user.hasEnteredInfo() ? (
+          <a onClick={this.toggleProfileForm}>Edit Profile</a>
+        ) : (
+          this.renderOnboarding()
+        )}
+      </div>
+    );
+  }
+
+  private renderOnboarding() {
+    const { profileFormVisible, whyProfileVisible } = this.state;
+    return [
+      !profileFormVisible && (
+        <button
+          type="button"
+          id="create-profile-button"
+          onClick={this.toggleProfileForm}>
+          Create a profile
+        </button>
+      ),
+      <div>
+        {whyProfileVisible ? (
+          <div>
+            <span id="why-profile-title">Why a profile?</span>
+            <div id="why-profile">
+              <p id="why-profile-text">
+                Copy explaining value of profile &amp; demographic capture:
+                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla
+                id orci dui.
+              </p>
+              <p>
+                <a name="" onClick={this.toggleWhyProfile}>
+                  Close
+                </a>
+              </p>
+            </div>
+          </div>
+        ) : (
+          <span id="why-profile-title">
+            <a onClick={this.toggleWhyProfile}>Why a profile?</a>
+          </span>
+        )}
+      </div>,
+    ];
```

Comment:
why return an array with one element?

Reply:
It's two elements: the create-button and the why text/toggle

Reply:
well, that has very poor readability then :(

can you arrange this in a nicer way?

Reply:
tried to make it more readable by factoring the WhyText into a separate comp. LMKWYT

Reply:
definitely better.

i'd still prefer to keep the expanded and collapsed variants together, e.g. like this:

```
      <div>
        <span id="why-profile-title">
          {expanded ? (
            <a onClick={this.toggleExpanded}>Why a profile?</a>
          ) : (
            <span>Why a profile?</span>
          )}
        </span>
        {expanded && (
          <div id="why-profile">
            <p id="why-profile-text">
              Copy explaining value of profile &amp; demographic capture:
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla
              id orci dui.
            </p>
            <p>
              <a onClick={this.toggleExpanded}>Close</a>
            </p>
          </div>
        )}
      </div>
```

where `expanded` is a different name for `whyProfileVisible`.

Ideally, we'd also deduplicate the "Why a Profile?" text, but i've not tried that yet. The point of this is that if someone later comes and changes the copy, structure or style of the expanded or collapsed variant, they should be able to easily find the other variant as well.

The example above is also a non-CSS-visibility-switching solution, as you said you prefer in another comment (and the reasoning of which i can see).

Reply:
Makes sense to me 👍 Moved it into its own component (naming is a bit shortened, not out of disrepect but because of me not checking the comment again while making the change).